site/app/views/HomePageView.php

```diff
@@ -14,167 +14,38 @@ class HomePageView extends AbstractView {
-                                        $header = "<h3>Instructor:</h3>";
-                                break;
-                                case 2:
-                                        $header = "<h3>Full Access Grader:</h3>";
-                                break;
-                                case 3:
-                                        $header = "<h3>Grader:</h3>";
-                                break;
-                                default:
-                                        $header = "<h3>Student:</h3>";
-                                break;
-                        }
-
-                            $return .= <<<HTML
-                        <tr>
-                            <td colspan="8">
-                                {$header}
-                            </td>
-                        </tr>
-HTML;
-
-                        for($q = 0; $q < count($rankWithCourse[$i]); $q++){
-                            $display_text = $rankWithCourse[$i][$q]->getSemester() . " " . $rankWithCourse[$i][$q]->getTitle();
-                            if($rankWithCourse[$i][$q]->getDisplayName() !== "") {
-                                $display_text .= " " . htmlentities($rankWithCourse[$i][$q]->getDisplayName());
-                            }
-
-                        $return .= <<<HTML
-
-                        <tr>
-                            <td style="width:85%" colspan="30">
-                                <a id="{$rankWithCourse[$i][$q]->getSemester()}_{$rankWithCourse[$i][$q]->getTitle()}" class="btn btn-primary btn-block" style=" white-space: normal;" href="{$this->core->buildUrl(array('component' => 'navigation', 'course' => $rankWithCourse[$i][$q]->getTitle(), 'semester' => $rankWithCourse[$i][$q]->getSemester()))}"> {$display_text}{$user->accessAdmin()}</a>
-                               
-                            </td>
-                        </tr>
-HTML;
-                        }
-
-                        $pos++;
-                    }
 
+        //Assemble courses into rank lists
+        foreach ($courses as $course) {
+            $rank = $this->core->getQueries()->getGroupForUserInClass($course->getSemester(), $course->getTitle(), $user->getId());
+            array_push($ranks[$rank]["courses"], $course);
+        }
 
-                    $return .= <<<HTML
-                    </tbody>
-                </table>
-            </div>
-        </div>
-    </div>
-</div>
-HTML;
-        return $return;
+        //Filter any ranks with no courses
+        $ranks = array_filter($ranks, function($rank) {
+            return count($rank["courses"]) > 0;
+        });
+
+        return $this->core->getOutput()->renderTwigTemplate('HomePage.twig', [
+            "user" => $user,
+            "ranks" => $ranks,
+            "changeNameText" => $changeNameText,
+            "showChangePassword" => $this->core->getAuthentication() instanceof DatabaseAuthentication
+        ]);
```

Comment:
Do we want to even use these views anymore? It seems like we'd want to move any of this functionality to controllers (or create Services or something) and just try and delete files out of this directory.